### PR TITLE
[server] engine option should accept an object

### PR DESCRIPTION
**Pull request by james-tindal**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

From [documentation on adding engines](https://serverjs.io/documentation/options/#writing-your-own-engine):
```js
const options = {
  engine: {
    // Register two keys for the same render function
    nunjucks,
    njk: nunjucks
  }
};

server(options, [...])
```

It has been this way since 2017 ([commit](https://github.com/franciscop/server/commit/bb329fd9bed9ec91bff333e55487566db2f2d592))

### CODE REVIEW

File: types/server/typings/options.d.ts [hunk 1]
```diff
@@ -42,12 +42,15 @@ export interface CsurfOptions {
     sessionKey?: string | undefined;
 }
 
+export {}
```
Comment (DazSanchez) [inline types/server/typings/options.d.ts:45]: This shouldn't be here.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## server ([<kbd>unpkg</kbd>](https://unpkg.com/browse/server@latest/))
was missing the following properties:
1. utils
1. plugins
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 59cb61b09a56ba328ca703ac0dd3ffe7ed7ff0b0
</p>

**Review by DazSanchez — CHANGES_REQUESTED** [review]
I'd encourage you to add a test scenario on the [test file](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/server/server-tests.ts) for your change (could be the example on the docs for this feature) and then run `npm test server` so we are sure that there's no problem.

File: types/server/typings/options.d.ts (hunk 1 above)
Comment (james-tindal) [reply types/server/typings/options.d.ts:45]: @typescript-bot told me to put it there. Otherwise EngineArgs is exported automatically.

Comment (james-tindal) [reply types/server/typings/options.d.ts:45]: Could add a comment explaining why it's there?

**Review by MysteryBlokHed — APPROVED** [review]
LGTM!

**Review by DazSanchez — APPROVED** [review]

**Comment by james-tindal** [comment]
Ready to merge

**Pull request merged**